sequence_processing_pipeline/Pipeline.py

```diff
@@ -553,55 +593,62 @@ def generate_sample_info_files(self, addl_info=None):
-                    x in self.sample_sheet.samples]
-            df = pd.DataFrame(data, columns=['sample_name', 'project_name'])
-
-        if addl_info is not None:
-            df = pd.concat([df, addl_info],
-                           ignore_index=True).drop_duplicates()
+            controls = self.sample_sheet.get_denormalized_controls_list()
+            df = pd.DataFrame(controls)
 
-        df = df[df["sample_name"].str.startswith("BLANK") == True]  # noqa
-        samples = list(df.to_records(index=False))
-        projects = df.project_name.unique()
+        projects = df[PROJECT_FULL_NAME_KEY].unique()
 
         paths = []
         for project in projects:
-            samples_in_proj = [x for x, y in samples if y == project]
-            some_path = join(self.output_path,
-                             f'{self.run_id}_{project}_blanks.tsv')
-            paths.append(some_path)
-            with open(some_path, 'w') as f:
-                # write out header to disk
-                f.write('\t'.join(Pipeline.sif_header) + '\n')
-
-                # for now, populate values that can't be derived from the
-                # sample-sheet w/'EMPTY'.
-                for sample in samples_in_proj:
-                    row = {}
-                    for column, default_value in zip(Pipeline.sif_header,
-                                                     Pipeline.sif_defaults):
-                        # ensure all defaults are converted to strings.
-                        row[column] = str(default_value)
-
-                    # overwrite default title w/sample_project name, minus
-                    # Qiita ID.
-                    row['title'] = sub(r'_\d+$', r'', project)
-
-                    # generate values for the four columns that must be
-                    # determined from sample-sheet information.
-
-                    # convert 'BLANK14_10F' to 'BLANK14.10F', etc.
-                    row['sample_name'] = sample.replace('_', '.')
-                    row['host_subject_id'] = sample.replace('_', '.')
-                    row['description'] = sample.replace('_', '.')
-                    row['collection_timestamp'] = self.get_date_from_run_id()
-
-                    row = [row[x] for x in Pipeline.sif_header]
-                    f.write('\t'.join(row) + '\n')
+            project_info = parse_project_name(project)
+
+            curr_fname = self.make_sif_fname(self.run_id, project)
+            curr_fp = join(self.output_path, curr_fname)
+            paths.append(curr_fp)
+
+            controls_in_proj_df = \
+                df.loc[df[PROJECT_FULL_NAME_KEY] == project].copy()
+
+            # TODO: remove this loop and replace with spp_metadata call at end
+            for column, default_value in zip(Pipeline.sif_header,
+                                             Pipeline.sif_defaults):
+                # ensure all defaults are converted to strings.
```

Comment:
What about representing them as `str` in `sir_defaults` rather than casting?

Reply:
I can see that this could be a matter of opinion.  My take on it is (a) by casting, I only have to remember to make stuff into strings in ONE place, rather than every time I add or change something in the defaults, and (b) I'd rather check for which columns not to fill by testing against `None` than against "None".  All that said, this is part of what's going to have to get ripped out and replaced to support the SPP handling *multiple* different kinds of controls rather than control blanks, so I don't want to give it too much attention :)

Reply:
Ok thanks. If this is just going out through `to_csv`, is it necessary to cast at all given that the csv won't contain type information?

Reply:
My experience with `pandas` is that if it decides via its automatic typing that, e.g. a given column contains floats, then when it writes to csv it will take, say, an input given as `193` and decide to write it out as `193.0` in the csv file.  I hate that :). Telling it everything is a string short-circuits any such overly-helpful formatting shenanigans.